packages/next/src/build/webpack-config.ts

```diff
@@ -1954,6 +1954,12 @@ export default async function getBaseWebpackConfig(
                 ),
                 layer: WEBPACK_LAYERS.metadataRoute,
               },
+              {
+                // Ensure that the app page module is in the client layers, this
+                // enables React to work correctly for RSC.
+                layer: WEBPACK_LAYERS.client,
```

Comment:
This does not seem to correct to me 🤔  cc @shuding

Reply:
seems right to me though I propose we rename this layer to ssr to clarify it is the client bundle running on the server

Reply:
or is the thing that you are saying is wrong that this should be the issuerlayer?

Reply:
@gnoff is right, the `client` layer is technically the SSR layer. A rename would be helpful, I can swap it out in a separate PR

Reply:
Yep agreed that it's better to rename `.server` and `.client` layers to be `.rsc` and `.ssr`, perhaps.

Reply:
Yeah let's rename them 👍

Reply:
Renaming in a follow-up sounds ideal to me to keep changes isolated